sdk/search/Azure.Search/README.md

```diff
@@ -0,0 +1,171 @@
+
+## Getting started
+
+### Install the package
+
+Install the Azure Cognitive Search client library for .NET with [NuGet][nuget]:
+
+```Powershell
+dotnet add package Azure.Search --version 1.0.0-preview.1
+```
+
+### Prerequisites
+
+You need an [Azure subscription][azure_sub] and a
+[Search Service][create_search_service_docs] to use this package.
+
+To create a new Search Service, you can use the [Azure Portal][create_search_service_docs],
+[Azure PowerShell][create_search_service_ps], or the [Azure CLI][create_search_service_cli].
+Here's an example using the Azure CLI to create a free instance for getting started:
+
+```Powershell
+az search service create --name mysearch --resource-group mysearch-rg --sku free --location westus
+```
+
+## Key concepts
+
+Azure Cognitive Search is the only cloud search service with built-in AI
+capabilities that enrich all types of information to easily identify and
+explore relevant content at scale.
+
+- Fully managed search as a service to reduce complexity and scale easily
+- Auto-complete, geospatial search, filtering, and faceting capabilities for a rich user experience
+- Built-in AI capabilities including OCR, key phrase extraction, and named entity recognition to unlock insights
+- Flexible integration of custom models, classifiers, and rankers to fit your domain-specific needs
+
+`Azure.Search` offers two types of resources:
+
+- The _search service_ used via `SearchServiceClient` to create and manage search indexes
+- A _search index_ in the search service used via `SearchIndexClient` to query and manage the documents in a search index
+
+## Examples
+
+### Getting service statistics
+
+```C# Snippet:Azure_Search_Tests_Samples_GetStatisticsAsync
+// Create a new SearchServiceClient
+Uri endpoint = new Uri(Environment.GetEnvironmentVariable("SEARCH_ENDPOINT"));
+SearchApiKeyCredential credential = new SearchApiKeyCredential(
+    Environment.GetEnvironmentVariable("SEARCH_API_KEY"));
+SearchServiceClient search = new SearchServiceClient(endpoint, credential);
+
+// Get and report the Search Service statistics
+Response<SearchServiceStatistics> stats = await search.GetStatisticsAsync();
+Console.WriteLine($"You are using {stats.Value.Counters.IndexCounter.Usage} of {stats.Value.Counters.IndexCounter.Quota} indexes.");
+```
+
+### Counting the documents in an index
+
+```C# Snippet:Azure_Search_Tests_Samples_GetCountAsync
+// Create a SearchIndexClient
+Uri endpoint = new Uri(Environment.GetEnvironmentVariable("SEARCH_ENDPOINT"));
+SearchApiKeyCredential credential = new SearchApiKeyCredential(
+    Environment.GetEnvironmentVariable("SEARCH_API_KEY"));
+string indexName = Environment.GetEnvironmentVariable("SEARCH_INDEX");
+SearchIndexClient index = new SearchIndexClient(endpoint, indexName, credential);
+
+// Get and report the number of documents in the index
+Response<long> count = await index.GetCountAsync();
```

Comment:
Guidelines changed there as well. There are some cases where we might build and return a "subclient" - like we do in Storage - but it's not often. Instead, methods might use a "scope" of that nature as part of the name - like we often did in Key Vault. So, for example, if Count makes sense for other things besides Documents, we might have something like `GetDocumentCound(Async)()`.

Reply:
@heaths Makes sense. `SearchIndexClient` sort of is a sub-client of `SearchServiceClient`, but that's a separate issue.

Given that the REST API may have a new resources besides "docs" in the future, it's very important to name this `GetDocumentCount` now, because otherwise:
1. The name becomes ambiguous in the future, and changing it would be a breaking change
1. It's already not obvious what is being counted.

Reply:
/cc @tg-msft

Reply:
I think that makes sense.  I'll change this to `GetDocumentCount`, change lookup to `GetDocument`, etc.

Reply:
(But let's also chat about this with the Arch Board tomorrow - we've not done the operation grouping from Track 1 and instead tended to put those in different clients.  It's worth considering the possibility of a SearchDocumentClient, etc., if we want to think about how these APIs might grow over the coming years.)

Reply:
I'm not a fan of one client per REST resource. Resources are a dime a dozen in our service. 😉

Reply:
Yep - but I also don't want 500 methods hanging off of SearchServiceClient for Indexes, Skillsets, Synonyms, etc.  We should talk about whether we want to consider extra clients, consider grouping operations a la Track 1, etc., to make sure we've got a plan for you to continue adding APIs for years to come.

Reply:
The Java SDK already has a gazillion methods on `SearchServiceClient`, so the sooner we figure this out, the better.